analyzer/src/update-gh.ts

```diff
@@ -0,0 +1,86 @@
+require("dotenv").config();
+import { readFileSync, writeFileSync } from "fs";
+
+import Providers from "./providers";
+import { getMonthIdentifier } from "./helper";
+import { Artifact } from "./types";
+import providers from "./providers";
+
+(async () => {
+
+    // get current month data
+    let currentTrendsJson: any = {};
+    let currentAllItemsJson: any = {};
+    try {
+        const filecontentTrends = readFileSync(
+            `${__dirname}/../../frontend/trends/webapp/model/trends.json`,
+            "utf8"
+        );
+        currentTrendsJson = JSON.parse(filecontentTrends);
+    } catch (e) {
+        if (e.code !== "ENOENT") {
+            throw e;
+        }
+    }
+    try {
+        const filecontentAllItems = readFileSync(
+            `${__dirname}/../../frontend/trends/webapp/model/allItems.json`,
+            "utf8"
+        );
+        currentAllItemsJson = JSON.parse(filecontentAllItems);
+    } catch (e) {
+        if (e.code !== "ENOENT") {
+            throw e;
+        }
+    }
+
+    // get data from github
+    const artifactsGithub: Artifact[] = await Providers[2].get(currentTrendsJson)
```

Comment:
This seems a bit dangerous to me as it would break if someone changes the order of the array elements. Please change it to a direct import of the GH provoider

Reply:
You are absolutely right here.
This is very risky and I should have changed it from the start.